modules/benchmark/R/format_wide2long.R

```diff
@@ -0,0 +1,103 @@
+##' @name format_wide2long
+##' @title format_wide2long
+##' 
+##' @param out wide format data
+##' @param format as returned by query.format.vars
+##' @param vars_used 
+##' @param time.row
+##' @param vars.used.index
+##' @return list of updated values
+##' @export
+##' @author Istem Fer
+##' Function to convert wide format to long format
+format_wide2long <- function(out, format, vars_used, time.row, vars.used.index){
+  
+  # which cols are "wide"
+  dindx <- vars_used$bety_name == unique(vars_used$bety_name[duplicated(vars_used$bety_name)])
+  wide.vars <- vars_used$input_name[dindx]
+  long.vars <- colnames(out)[!(colnames(out) %in% wide.vars)]
+  mout      <- reshape2::melt(out, id = long.vars) 
+  
+  # GapMacro example:
+  # suppose that the wide variable is DBH, mout would look something like this:
+  ####################
+  # variable  value  #  
+  #  diam12     12   #
+  #  diam14     10   #
+  #  diam16     23   #
+  ####################
+  
+  # diam12/14/16 are the column headers in the input file
+  # you probably want to rename "value" column as "DBH" here
+  colnames(mout)[colnames(mout) == "value"] <- unique(vars_used$bety_name[duplicated(vars_used$bety_name)])
```

Comment:
I thought about this too but one wide-variable was my only test case so far and this was a first pass :)  I'll see if the same or a similar approach will hold up with multiple wide variables in a single dataset